src/quicklist.c

```diff
@@ -147,6 +149,15 @@ void quicklistSetCompressDepth(quicklist *quicklist, int compress) {
     quicklist->compress = compress;
 }
 
+/* Calculate the size limit of the quicklist node based on negative 'fill'. */
+size_t quicklistNodeNegFillLimit(int fill) {
```

Comment:
we can declare it in advance at the top of this file to avoid moving it to here.